### PR TITLE
Run branch-protector Prow job for Knative org

### PR DESCRIPTION
**Pull request by chizhg**
**What this PR does, why we need it**:
We had been running `branch-protector` job for `knative-sandbox` org for a while, and no issues were reported, so enable it for `knative` org as well.

Note the previous Prow job was mistakenly configured to run on one of my test branch, and one day I deleted it, so the Prow jobs had been failing for a while due to `branch cannot be found` error. I have fixed it in this PR.

/cc @mattmoor

### CODE REVIEW

**Review by mattmoor — COMMENTED** [review]
/lgtm

**Comment by mattmoor** [comment]
cc @knative/knative-release-leads

File: config/prod/prow/jobs/custom/branchprotector.yaml [hunk 1]
```diff
@@ -15,34 +15,35 @@
...
-        volumeMounts:
-        - name: oauth
-          mountPath: /etc/github
-          readOnly: true
-      volumes:
-      - name: oauth
-        secret:
-          secretName: github-token-for-branchprotector
+# Do not enable presubmit tests for now since it's taking extra long time.
+#presubmits:
```
Comment (mattmoor) [inline config/prod/prow/jobs/custom/branchprotector.yaml:19]: Is there an issue to fix this?

Comment (chizhg) [reply config/prod/prow/jobs/custom/branchprotector.yaml]: Thanks for the reminder, just created the issue - https://github.com/knative/test-infra/issues/2559

**Review by mattmoor — APPROVED** [review]
/lgtm

**Pull request merged**